### ISSUE TITLE
ratatouille.lv2 ain´t building

### ISSUE THREAD

**Issue by ghost** [brummer10/Ratatouille.lv2#29]
The packager of the AUR package of ratatouille.lv2 for Arch Linux explained to me that the build error is very rarely happening from time to time - here the build process is never finishing successfully.
The error message is:
build/ratatouille.lv2/PKGBUILD: line 28: 2003 Aborted lv2lint -M pack -s '_Z*' -I bin/$_name "$_plugin_uri" ==> ERROR: A failure occurred in check(). Aborting... Build failed with exit code 4

I have  reported the issue on AUR page and also on the git repository of the packager:
https://github.com/osam-cologne/archlinux-proaudio/issues/456
Artix Linux system here, with 6.12 kernel - system is always up2date.
When disabling lv2lint checking, the build process is finishing successfully.

**Comment by SpotlightKid** [comment]
To explain a bit more. I believe, this is not a build error per se, but the plugin fails to instantiate when checking it with `lv2lint` but segfaults.

**Comment by brummer10** [comment]
There is not much I could do about that.

**Comment by ghost** [comment]
thanks for looking at it again. It is no big issue, because the lint checking can be disabled easily in the makepkg file.
Nevertheless this is a lil annoying:
Maybe other users  are asking for your support because of this and this is not time efficient, then.
Also, I am batch upgrading my AUR packages so this is annoying for me because I have to edit the makepkg ffile everytime a new update is available.
Ratatouille is an amazing plugin and everything is ok, just saying...
There is always a nasty issue somewhere!

**Comment by SpotlightKid** [comment]
Wasn't this fixed via https://github.com/osam-cologne/archlinux-proaudio/pull/542 ?

**Comment by brummer10** [comment]
@TeamB12 
As I pointed out earlier, there is nothing much I could do about that. 
When I run lv2lint here I got:

```
v2lint 0.16.2
Copyright (c) 2016-2021 Hanspeter Portner (dev@open-music-kontrollers.ch)
Released under Artistic License 2.0 by Open Music Kontrollers
<urn:brummer:ratatouille>
Missing feature options.
    [WARN]  Plugin Author Email
              foaf:mbox not found
              seeAlso: <http://lv2plug.in/ns/lv2core#project>
    [WARN]  Plugin Author Homepage
              foaf:homepage not found
              seeAlso: <http://lv2plug.in/ns/lv2core#project>
    [WARN]  Plugin Options Feature
              opts:{required,supported} options not defined
              seeAlso: <http://lv2plug.in/ns/ext/options#supportedOption>

```


Those are just warnings, and, the last one 'Plugin Options Feature' looks wrong to me, as it disappear when I remove the definition from the ttl file. 

However, later lv2lint versions from gitlab fail with several reasons, but, keep in mind there is a reason why never a newer version of lv2lint was released. So using a later version is on your own risk. 
Then, when lv2lint tries to do a X11 window test, it does that terrible wrong. It try to open the plugin UI without providing a parent Window. (A parent window is needed by the specs). Ratatouille checks by itself if a given parent window is valid, if not, it clean up itself, close it self and  notice the host that it wouldn't run. lv2lint mark that as a error, but, it is there for house keeping and is absolutely right.

**Comment by ghost** [comment]
thanks to you all. thanks also for this explanation.
You are super fast - you don´t have to be that fast - thanks.
Now I know and I just stay cool. Let's cross fingers that no other  lv2 plugin will show the same issues, one day.
